### DISCUSSION TITLE
nvidia-settings DO command does nothing

### DISCUSSION THREAD

**Discussion by micoro** [LizardByte/community#320] [Sunshine] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

Placing the following line as DO command (in application or in configuration) or as detach command does not have any effect:
`flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"`

Tested with flatpak and appImage.

### Expected Behavior

The monitor should change to output at resolution of 2560x1440.

### Additional Context

I want to change resolution of my main monitor: it is a widescreen (3440x1440) but my TV is 16:9 so I need to change the dimension from 3440x1440 to 2560x1440. 

xrandr is not an option because my monitor does not list 2560x1440 as an option and any tentative I have done with it fails.

nvidia-settings does work changing the output resolution to 2560x1440. Tested multiple times from terminal, also while running sunshine and moonlight from the TV: everything works fine.
The problem is that if I place that very same string in the DO command it does not work.

### Host Operating System

Linux

### Operating System Version

Manjaro - linux kernel 6.5

### Architecture

64 bit

### Sunshine commit or version

0.20.0

### Package

Linux - flatpak

### GPU Type

Nvidia

### GPU Model

GeForce 3070 Ti

### GPU Driver/Mesa Version

535.113.01

### Capture Method (Linux Only)

X11

### Config

```Shell
global_prep_cmd = [{"do":"flatpak-spawn --host nvidia-settings --assign CurrentMetaMode=\"DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}\"","undo":""}]
output_name = 1
audio_sink = sink-sunshine-stereo
nvenc_preset = 2
min_log_level = 1
capture = x11
```

### Relevant log output

```shell
[2023:10:12:19:55:09]: Debug: Parsed executable [flatpak-spawn] from command [flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"]
[2023:10:12:19:55:09]: Info: Executing Do Cmd: [flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"]
[2023:10:12:19:55:09]: Debug: Parsed executable [flatpak-spawn] from command [flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"]
[2023:10:12:19:55:09]: Info: Spawning [flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"] in ["/usr/bin"]
[2023:10:12:19:55:09]: Debug: Parsed executable [flatpak-spawn] from command [flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"]
[2023:10:12:19:55:09]: Info: Spawning [flatpak-spawn --host nvidia-settings --assign CurrentMetaMode="DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}"] in ["/usr/bin"]



First two lines are the log in case of the DO command from configuration
The last four lines are the log from the DO and detach command from application
```

**Comment by ReenigneArcher** [comment]
You might need to prefix the command with `sh -c` as in our examples. https://docs.lizardbyte.dev/projects/sunshine/en/nightly/about/guides/app_examples.html#changing-resolution-and-refresh-rate-linux-x11

Not sure how you combine that with the `flatpak-spawn --host` prefix though.

**Reply by micoro** [reply]
tried with different combinations: 
`flatpak-spawn --host sh -c "nvidia-settings --assign CurrentMetaMode=\"DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}\""`

`sh -c "flatpak-spawn --host nvidia-settings --assign CurrentMetaMode=\"DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}\""`

and I installed the AUR package to try it on without the `flatpak-spawn`

`sh -c "nvidia-settings --assign CurrentMetaMode=\"DP-0: 3440x1440_144 +0+0 {viewportout=2560x1440+440+0}\"`


All of them fail with `code 127`.

(from terminal it works without problem)

**Reply by ReenigneArcher** [reply]
Exit code 127: when the given command is not found within your PATH system variable

**Reply by micoro** [reply, marked as answer]
SOLVED

I tried multiple options and in the end what worked was to wrap my nvidia-settings function inside a new .sh file and call that file.

`sh -c /home/xxx/.screenlayout/steamLinkTV.sh`

Probably there could have been some leaner solution and I did something wrong, but this works.

**Reply by ehabajmal** [reply]
Funnily enough, I ran into the same issue just yesterday and came to the same conclusion (wrap in .sh and call it). PR here is to include it in the docs: https://github.com/LizardByte/Sunshine/pull/1767
